lib/cc/cli/config_generator.rb

```diff
@@ -80,13 +83,24 @@ def workspace_files
           if non_excluded_paths.empty?
             []
           else
-            find_cmd = "find #{non_excluded_paths.map(&:shellescape).join(' ')} -type f -print0"
-            `#{find_cmd}`.strip.split("\0").map do |path|
-              path.sub(%r{^\.\/}, "")
-            end
+            find_workspace_files
           end
         end
       end
+
+      def find_workspace_files
+        find_cmd = %w[find] + non_excluded_paths + %w[-type f -print0]
+        pid, _, out, err = POSIX::Spawn.popen4(*find_cmd)
+        _, status = Process.waitpid2(pid)
+
+        if status.success?
+          out.read.strip.split("\0").map do |path|
+            path.sub(%r{^\.\/}, "")
+          end
+        else
+          raise FindFailError, "We ran '#{find_cmd.shelljoin}' to find analyzeable files, but the command failed with this error: '#{err.read}'"
```

Comment:
I favor the second, at least for the time being. I think wrapping up our spawn logic somewhere might be useful, but I think getting better user-facing errors out is important, so I'll focus on that for now.